driver-core/src/main/com/mongodb/assertions/Assertions.java

```diff
@@ -119,6 +126,57 @@ public static void doesNotContainNull(final String name, final Collection<?> col
         }
     }
 
+    /**
+     * @param value A value to check.
+     * @param <T> The type of {@code value}.
+     * @return {@code null}.
+     * @throws AssertionError If {@code value} is not {@code null}.
+     */
+    @Nullable
+    public static <T> T assertNull(@Nullable final T value) throws AssertionError {
```

Comment:
None of the other methods throw checked exceptions. Should these follow the existing convention?

Reply:
Do you mean unchecked?  I would agree that there is no need for the throws clause but it also doesn't bother me given that throwing this exception is the whole purpose of the method.

Reply:
I use the `throws` clause to document thrown exceptions that are part of a method contract (even if they are unchecked) in situations when I don't write javadocs. But then when I write javadocs, I still specify the `throws` clause (to have the same approach everywhere) even though it is duplicated with the `@throws` tag. Since there does not seem to be a strong push for removing the clause, I left it.